### DISCUSSION TITLE
Why do TCP listener and TCP sender connections get disconnected?

### DISCUSSION THREAD

**Discussion by Coooz** [nextgenhealthcare/connect#6051] [Q&A]
Hi,

The below images show parts of the connection log of two channels that are running at the site of a customer of ours, sorted by timestamp, most recent on top. I wonder what could be the cause of the source and destination getting disconnected. Although apparently the connection gets restored automatically each time, the customer complains about messages queueing up. 
What needs to be done to stop the connectors from disconnecting?

![image](https://github.com/nextgenhealthcare/connect/assets/88899312/495801af-d488-4273-b53b-7a50a6193df7)

![image](https://github.com/nextgenhealthcare/connect/assets/88899312/33f2b4de-058e-4206-813f-cc87e1c5509c)

Thanks!

**Comment by pacmano1** [comment]
What are the settings on the listeners and senders?  You may not be waiting long enough and closing the connection to soon.  Also, is your channel set to keep the connection open?

**Comment by Coooz** [comment]
Thanks for replying @pacmano1. The reason for messages queueing up appeared to be poor perfomance of a SQL data centre instead of connectors regularly disconnecting. The latter is not a cause of concern. This ticket can be closed.
